File: AddressBook/AddressBook/Dtos/PhoneLabel.cs
using AddressBook.Common.Paging;
using System.ComponentModel.DataAnnotations;

namespace AddressBook.Dtos
{
    public class PhoneLabelRes
    {
        public int PhoneLabelId { get; set; }
        public string? Label { get; set; }
    }

    public class PhoneLabelReqEdit
    {
        [Required, MaxLength(20)]
        public string? Label { get; set; }
    }

    public class PhoneLabelReqSearch : PagedReq
    {

    }
}

File: AddressBook/react-client/src/dtos/PhoneLabel.ts
import Common from "../utility/Common";
import { PagedReq } from "./PagedReq";

export interface PhoneLabelRes {
  phoneLabelId?: number;
  label?: string;
}

export class PhoneLabelReqEdit {
  label?: string = "";
}

export class PhoneLabelReqSearch extends PagedReq {
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    {}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
  }
}